### PR TITLE
Gamified exercies frontend

### PR DESCRIPTION
**Pull request by Andazu**
## Description

[App Course] [Frontend] "As a waste picker, I want to be able to review the exercises and get feedback from it"

## Changes

- Changed the review answer text into a button under the scrollview instead
- Made the confirm answer button change opacity when no answer has been selected
- Made it possible to press the text of the answer and not only the radio button
- Added background color to the scrollview dependant on correct or incorrect answers
- Added background color to the feedback text dependant on correct or incorrect answers
- Added icons to the feedback dependant on correct or incorrect answers
- Changed all tailwind values to percentage instead of pixels for responsiveness
- Updated the progressbar component to handle input parameters to change dimensions
- Various visual fixes

## Related Issues

Perhaps not enough tests, but the one included tests if the screen is rendered correctly

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

Screenshot_1697447481

## Notes for Reviewers

When using the new progressbar component, it is possible to change its size to using this syntax:
```
<CustomProgressBar progress={%} width={%} height={%}></CustomProgressBar>

*REPLACE THE % WITH VALUE IN PERCENTAGE

```

### CODE REVIEW

**Comment by Jatewo** [comment]
Tests are failing. Are you sure you want a review now, or do you have some changes to do beforehand?

File: eml/.gitignore [hunk 1]
```diff
@@ -9,6 +9,8 @@ npm-debug.*
 *.orig.*
 web-build/
 eslint-demo
+package-lock.json
+package.json
```
Comment (JackSMunn) [inline eml/.gitignore:13]: STOP. HOLD. VENT. 

package.json, og package-lock.json

disse 2 filer skal trackes af git, ellers når i updater eller sætter dependencies ind, så får vi andre det ikke når vi kører  npm i

File: eml/components/exercise/ExerciseButtons.jsx [hunk 2]
```diff
@@ -48,14 +40,6 @@ export default function ExerciseButtons({ exerciseId, answers, sectionId, course
...
   const [button, setButton] = useState(true)
 
-  function handlePlaySound(tts) {
-    Speech.stop()
-    Speech.speak(tts, {
-      language: "portuguese",
-      voice: "com.apple.ttsbundle.Luciana-compact"
-    }
-    )
-  }
```
Comment (JackSMunn) [inline eml/components/exercise/ExerciseButtons.jsx:58]: Hvorfor bliver dette slettet?

File: eml/components/exercise/LeaveButton.js [hunk 3]
```diff
@@ -20,17 +20,17 @@ const LeaveButton = ({ navigationPlace, courseId }) => {
       }
       icon={
         <Icon
-          size={40}
+          size={25}
           name="chevron-left"
           type="material-community"
-          color="rgb(75,85,99)"
+          color="#383838"
```
Comment (JackSMunn) [inline eml/components/exercise/LeaveButton.js:26]: It is necessary to use stylesheets due to some reason?

If not, then integration team has agreed that all use NativeWind.

File: eml/components/exercise/LeaveButton.js [hunk 4]
```diff
@@ -20,17 +20,17 @@ const LeaveButton = ({ navigationPlace, courseId }) => {
...
     ></Button>
-  )
-}
+  );
+};
 const styles = StyleSheet.create({
-  buttons: { width: 50, height: 50 }
-})
+  buttons: { width: 50, height: 50 },
+});
```
Comment (JackSMunn) [inline eml/components/exercise/LeaveButton.js:34]: You are using stylesheets here

File: eml/components/exercise/Progressbar.js [hunk 5]
```diff
@@ -1,19 +1,31 @@
...
+        progress={progress / 100}
+        width={ScreenWidth * (width / 100)}
+        height={ScreenHeight * (height / 100)}
+        color="rgba(94, 204, 233, 1)"
+        unfilledColor="rgba(228, 242, 245, 1)"
+        borderWidth={0}
+        borderRadius={8}
+      ></Progress.Bar>
+      <Text className="px-5 text-center font-montserrat-bold text-caption-medium text-projectBlack">
+        {progress}%
```
Comment (JackSMunn) [inline eml/components/exercise/Progressbar.js:25]: Again, stylesheets and not NativeWind.

File: eml/screens/excercise/ExerciseScreen.jsx [hunk 6]
```diff
@@ -1,162 +1,185 @@
...
+    if (dummyExerciseData.answers[selectedAnswer - 1].isCorrect) {
+      setButtonClassName("bg-projectGreen");
+    } else {
+      setButtonClassName("bg-projectRed");
+    }
+    setShowFeedback(true);
+  }
+
+  /*async function getExercise() {
     const exercise = await StorageService.getNextExercise(sectionId);
```
Comment (JackSMunn) [inline eml/screens/excercise/ExerciseScreen.jsx:40]: remove if code is dead

File: eml/screens/excercise/ExerciseScreen.jsx [hunk 7]
```diff
@@ -1,162 +1,185 @@
...
+          sectionId: sectionId,
+        });
+      } else {
+        setExerciseData(exercise);
+      }
+    } else {
+      navigation.navigate("ErrorScreen");
     }
-    setExerciseData(exercise);
+  }*/
```
Comment (JackSMunn) [inline eml/screens/excercise/ExerciseScreen.jsx:56]: Remove if code is dead

File: eml/screens/excercise/ExerciseScreen.jsx [hunk 8]
```diff
@@ -1,162 +1,185 @@
...
+
+  async function getExercise() {
+    const exercise = dummyExerciseData;
+
+    if (exercise !== null) {
+      /* const courseId = exercise.courseId || "defaultCourseId";
+      const sectionId = exercise.sectionId || "defaultSectionId";*/
+
```
Comment (JackSMunn) [inline eml/screens/excercise/ExerciseScreen.jsx:64]: Remove if code is dead

File: eml/screens/excercise/ExerciseScreen.jsx [hunk 9]
```diff
@@ -1,162 +1,185 @@
...
+      if (exercise === true) {
+        /* navigation.navigate("SectionComplete", {
+          courseId: courseId,
+          sectionId: sectionId,
```
Comment (JackSMunn) [inline eml/screens/excercise/ExerciseScreen.jsx:68]: remove if code is dead

File: eml/screens/excercise/ExerciseScreen.jsx [hunk 10]
```diff
@@ -1,162 +1,185 @@
...
-      console.log(sections[i].isComplete);
+    for (let i = 0; i < sections.length; i++) {
+      console.log("Section is complete: ", sections[i].isComplete);
       if (sections[i].id === sectionId) {
         return sections[i].isComplete;
       }
     }
-    return null;
-  }
+  }*/
```
Comment (JackSMunn) [inline eml/screens/excercise/ExerciseScreen.jsx:89]: Remove if code is dead

**Review by JackSMunn — COMMENTED** [review]
You have deleted alot of tests, is this on purpose, this also includes reset-password tests. (which we have made I believe)

I have requested changes / clarification on stylesheets / nativewind.

**Pull request closed without merging**